src/collisions.fnl

```diff
@@ -0,0 +1,49 @@
+
+; There is so much optimisation that could be do here
+; even from just simple things like precalculating the l2distances
+; instead of theoretically having to compute it multiple times per loop
+; but I am tired and this needs to get done so there. 
```

Comment:
The big optimization is to use `love.physics`, which (without checking) is C/C++ code that should have adequate performance.